jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -211,30 +211,44 @@ public JibContainerBuilder createContainerBuilder(
         return JavaContainerBuilderHelper.fromExplodedWar(javaContainerBuilder, explodedWarPath);
       }
 
-      if (containerizingMode == ContainerizingMode.PACKAGED) {
-        String jarName = project.getBuild().getFinalName() + "." + project.getPackaging();
-        Path jar = Paths.get(project.getBuild().getDirectory(), jarName);
-        return javaContainerBuilder
-            .addDependencies(getDependencies())
-            .addToClasspath(jar)
-            .toContainerBuilder();
+      // Add dependencies
+      Set<Artifact> artifacts = project.getArtifacts();
+      List<Path> dependencies =
+          artifacts.stream().map(Artifact::getFile).map(File::toPath).collect(Collectors.toList());
+      javaContainerBuilder.addDependencies(dependencies);
+
+      switch (containerizingMode) {
+        case EXPLODED:
+          // Add resources, and classes
+          Path classesOutputDirectory = Paths.get(project.getBuild().getOutputDirectory());
+          Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+          javaContainerBuilder
+              .addResources(classesOutputDirectory, isClassFile.negate())
+              .addClasses(classesOutputDirectory, isClassFile);
+          break;
+
+        case PACKAGED:
+          // Add a JAR
+          String jarName = project.getBuild().getFinalName() + "." + project.getPackaging();
```

Comment:
So I think this should be `project.getArtifact().getFile()`, but if it doesn't exist then throw an error (_"project build result is missing, maybe you should run `package`"_)

Reply:
As we discussed offline, let's check `project.getArtifact().getFile()` first and fall back to this heuristic.

Reply:
I've expanded the logic to identify the JAR. PTAL.

Reply:
I think I'll revert using the Maven API (`getArtifact()` and `getAttachedArtifacts()`), as their results are non-deterministic. Rather, I think we should target the explicit `maven-jar-plugin` configuration. (In the future, we could consider checking other popular plugins.)